From 128611bdec030645dc0c6994ffe7fbdfc0022ff3 Mon Sep 17 00:00:00 2001
From: Ke Wu <mikewu@google.com>
Date: Mon, 30 Oct 2017 10:16:15 -0700
Subject: [PATCH] Change Docker log line max size to 100kB

---
 daemon/logger/copier.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemon/logger/copier.go b/daemon/logger/copier.go
index 10ab46e16..2de669cd5 100644
--- a/daemon/logger/copier.go
+++ b/daemon/logger/copier.go
@@ -10,7 +10,7 @@ import (
 )
 
 const (
-	bufSize  = 16 * 1024
+	bufSize  = 100 * 1024
 	readSize = 2 * 1024
 )
 
-- 
2.15.0.rc2.357.g7e34df9404-goog

