From 55f32166e3d65bab0023139882a46d14b540ea71 Mon Sep 17 00:00:00 2001
From: Xuewei Zhang <xueweiz@google.com>
Date: Wed, 22 May 2019 11:51:29 -0700
Subject: [PATCH] Add restart for containerd

---
 containerd.service | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/containerd.service b/containerd.service
index 1ae7fe86..db71c9aa 100644
--- a/containerd.service
+++ b/containerd.service
@@ -18,5 +18,8 @@ LimitNOFILE=infinity
 # Only systemd 226 and above support this version.
 TasksMax=infinity
 
+Restart=always
+RestartSec=10
+
 [Install]
 WantedBy=multi-user.target
-- 
2.21.0.1020.gf2820cf01a-goog

