From 3425202bb19e76a8ba3694688fc58ab0426f26b0 Mon Sep 17 00:00:00 2001
From: Aditya Kali <adityakali@google.com>
Date: Wed, 18 May 2016 17:12:58 -0700
Subject: [PATCH] use GO cross compiler

---
 Makefile | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 5de10d0..c63657b 100644
--- a/Makefile
+++ b/Makefile
@@ -15,14 +15,14 @@ all:
 ifneq ($(RUNC_LINK), $(wildcard $(RUNC_LINK)))
 	ln -sfn $(CURDIR) $(RUNC_LINK)
 endif
-	go build -ldflags "-X main.gitCommit=${COMMIT}" -tags "$(BUILDTAGS)" -o runc .
+	${GO} build -ldflags "-X main.gitCommit=${COMMIT}" -tags "$(BUILDTAGS)" -o runc .
 
 static:
-	CGO_ENABLED=1 go build -tags "$(BUILDTAGS) cgo static_build" -ldflags "-w -extldflags -static -X main.gitCommit=${COMMIT}" -o runc .
+	CGO_ENABLED=1 ${GO} build -tags "$(BUILDTAGS) cgo static_build" -ldflags "-w -extldflags -static -X main.gitCommit=${COMMIT}" -o runc .
 
 lint:
-	go vet ./...
-	go fmt ./...
+	${GO} vet ./...
+	${GO} fmt ./...
 
 runctestimage:
 	docker build -t $(RUNC_TEST_IMAGE) -f $(TEST_DOCKERFILE) .
@@ -32,7 +32,7 @@ test: runctestimage
 	tests/sniffTest
 
 localtest: all
-	go test -tags "$(BUILDTAGS)" ${TESTFLAGS} -v ./...
+	${GO} test -tags "$(BUILDTAGS)" ${TESTFLAGS} -v ./...
 
 dbuild: runctestimage
 	docker build -t $(RUNC_IMAGE) .
@@ -52,6 +52,6 @@ clean:
 
 validate:
 	script/validate-gofmt
-	go vet ./...
+	${GO} vet ./...
 
 ci: validate localtest
-- 
2.1.2

