commit 53723e4f32e1e75bb8ac207ba1f10a927375ae1c
Author: Amey Deshpande <ameyd@google.com>
Date:   Thu Mar 17 15:12:14 2016 -0700

    libsystemd-networkd: always use single-label hostname.

diff --git a/src/libsystemd-network/sd-dhcp-lease.c b/src/libsystemd-network/sd-dhcp-lease.c
index f5b9e22..f728a70 100644
--- a/src/libsystemd-network/sd-dhcp-lease.c
+++ b/src/libsystemd-network/sd-dhcp-lease.c
@@ -552,6 +552,7 @@ int dhcp_lease_parse_options(uint8_t code, uint8_t len, const uint8_t *option,
         {
                 _cleanup_free_ char *hostname = NULL;
                 char *e;
+                char *firstdot = NULL;
 
                 r = lease_parse_string(option, len, &hostname);
                 if (r < 0)
@@ -561,6 +562,19 @@ int dhcp_lease_parse_options(uint8_t code, uint8_t len, const uint8_t *option,
                 if (e)
                         *e = 0;
 
+                // Lakitu-specific behavior: always use single-label hostname
+                // or unqualified domain-name instead of FQDN.
+                // Lakitu configures systemd-networkd to set the hostname via
+                // DHCP. RFC 1497 says the DHCP server may send a
+                // fully-qualified domain name as the hostname.  The FQDN can
+                // be 255 bytes long, but the hostname on Linux is typically
+                // limited to 64 bytes.  To avoid dropping the FQDN as
+                // "invalid" for its length, we mutate |hostname| such that it
+                // contains only the single-label hostname.
+                if ((firstdot = strchr(hostname, '.')) != NULL) {
+                    *firstdot = 0;
+                }
+
                 if (!hostname_is_valid(hostname, false) || is_localhost(hostname))
                         break;
 
